dace/sdfg/utils.py

```diff
@@ -1046,3 +1046,82 @@ def unique_node_repr(graph: Union[SDFGState, ScopeSubgraphView],
     state = graph if isinstance(graph, SDFGState) else graph._graph
     return str(sdfg.sdfg_id) + "_" + str(sdfg.node_id(state)) + "_" + str(
         state.node_id(node))
+
+
+def update_path_subsets(
+    state: SDFGState,
+    inner_edge_info: Union[MultiConnectorEdge, nd.AccessNode],
+    new_subset: sbs.Subset,
+):
+    """
+    Will take the memlet path defined by :param inner_edge:, and recreate it with the specified new_subset
+    :param inner_edge_info: This is either the inner edge defining the memlet path that
+        should be modified, or an AccessNode with exactly one attached ingoing or outgoing
+        memlet path 
+    :param new_subset: The new subset set on the innermost edge
+    """
```

Comment:
I don't know, but it's definitvely a convenience function. Anyway here it's done via erasing and reading the memlet path. Would there be a cleaner way? Looking at the add_memlet path function, this looked quite complex

Reply:
Did you try just changing the innermost memlet and nothing else?

Reply:
What really it will just update? Anyway I'd like to keep parts of this function, since it also finds the innermost memlet and applies the change there given an AccessNode that has only one attached memlet path. That's definitively convenience, but I need it very often, and it simplifies codes quite often quite much

Reply:
Outer memlets do not have any functional meaning, they are only there to visualize data movement to the user. Memlet propagation takes care of making the outer annotations correct based on the inner annotations.

@alexnick83 How do you recommend doing this? Would you replace the edges, or change the inner memlet + call some propagation explicitly, or just change the inner edge and forget about it?

Reply:
Depends on the exact use case. We have something similar in RedundantArray (which needs some clean up). There we don't just replace a memlet but we potentially do some subset composition which memlet propagation cannot do without the extra information available in the transformation. Therefore, we just do the steps for the whole memlet path. I will have a look at the exact use case here and I will update you.